### ISSUE TITLE
mWeb - scan - When long pressing a section in mweb, the selection disappears within seconds

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#50011]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.42-0
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught during regression testing, add the test name, ID and link from TestRail:** N/A
**Issue reported by:** Applause - Internal Team

## Action Performed:
1. Launch both mweb and Android
2. Tap on a workspace chat
3. Tap plus icon - scan - upload an image using camera
4. Long press the description field

## Expected Result:
When long pressing a section, it must show same behavior in mweb and Android

## Actual Result:
When long pressing a section in mweb, the selection disappears within seconds but in Android selection remains

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [x] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

</details> 

https://github.com/user-attachments/assets/444e23fc-d374-4645-837b-5a516c1d0c37

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by lanitochka17** [comment]
@stephanieelliott  FYI I haven't added the External label as I wasn't 100% sure about this issue. Please take a look and add the label if you agree it's a bug and can be handled by external contributors

**Comment by huult** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.

When long pressing a section in mweb, the selection disappears within seconds 

### What is the root cause of that problem?

Long-touch triggers a contextmenu event, which ends the press and prevents the intended action from completing. This issue of [react-native-web](https://github.com/necolas/react-native-web/issues/2349)

### What changes do you think we should make in order to solve the problem?

To resolve this issue on mobile web, we can use `onPressOut` to handle the end of a press action.
```diff
// .src/components/MenuItem.tsx#L575
- onPressOut={ControlSelection.unblock}
+ onPressOut={(e) => {
+                                    if (Browser.isMobileChrome()) {
+                                        if (shouldCheckActionAllowedOnPress) {
+                                            Session.checkIfActionIsAllowed(onPressAction(e), isAnonymousAction);
+                                        } else {
+                                            onPressAction(e);
+                                        }
+                                   }

+                                    ControlSelection.unblock();
+                                }}
```

<summary>
<details>POC

https://github.com/user-attachments/assets/aa364ff0-cb70-44cc-ace1-10cfd0c6e844

</details>
</summary>

**Comment by c3024** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.

Long pressing a menu item on Android mobile web causes the pressed state feedback to disappear after a short time when the menu item is pressed.

### What is the root cause of that problem?

When no `onLongPress` handler is attached to a `Pressable` component, Android Chrome overrides the pressed state and activates the browser's default behavior. In this code:

[https://github.com/Expensify/App/blob/a6d812c09848d9d3748109ab775b47099742b8f9/src/components/Pressable/GenericPressable/BaseGenericPressable.tsx#L152](https://github.com/Expensify/App/blob/a6d812c09848d9d3748109ab775b47099742b8f9/src/components/Pressable/GenericPressable/BaseGenericPressable.tsx#L152)

the `onLongPress` handler is passed as `undefined` when the check `!isDisabled && onLongPress` evaluates to `false`, which is the case here.

### What changes do you think we should make in order to solve the problem?

We should pass a no-operation function (noop) in the required cases so that a handler, although noop, is set for `onLongPress`. By registering this handler, Android Chrome does not override the long press behavior. A similar fix is also discussed [here](https://github.com/necolas/react-native-web/issues/2349#issuecomment-1195564240).

We implemented a similar fix directly to the `BaseGenericPressable` [here](https://github.com/Expensify/App/pull/26931/) when we had an issue with long press on emojis. However, the changes to `BaseGenericPressable` had to be reverted [here](https://github.com/Expensify/App/pull/29601) because we overrode the browser's default behavior in all cases of using this `Pressable`, and `TextInput` also uses this `Pressable`. So, when long pressing on the text input, the browser's default menu of Copy/Paste/Select All, etc., was not displayed when we passed a default noop to the `onLongPress`.

I think `MenuItem`s do not need the browser's default behavior of long press in any case. So, we can pass a default noop for the long press in the `MenuItem` component like this:

```javascript
onSecondaryInteraction = Browser.isMobileChrome() ? () => {} : undefined,
```

here:

[https://github.com/Expensify/App/blob/a6d812c09848d9d3748109ab775b47099742b8f9/src/components/MenuItem.tsx#L418](https://github.com/Expensify/App/blob/a6d812c09848d9d3748109ab775b47099742b8f9/src/components/MenuItem.tsx#L418)

### What alternative solutions did you explore? (Optional)

We can just pass the noop for only the `MenuItem`s where we see this behavior, like we do in `MoneyRequestView` and `TripDetailsView`, if we find that the long press browser default context menu of Android Chrome is required for some `MenuItem`s.

**Comment by stephanieelliott** [comment]
Long press doesn't have any function in this scenario, so it seems this would not have any actual impact on users. Gonna close as `Not a priority`
